### PR TITLE
Add content on securing communication with the Gateway and the backend

### PR DESCRIPTION
**Pull request by Nashaath**
## Purpose
Add content on securing communication with the Gateway and the backend.

### CODE REVIEW

**Comment by Nashaath** [comment]
I’ve requested the marketing team to redraw the diagram according to the Choreo diagram guidelines. Once the redrawn diagram is shared, I’ll be replacing the existing with it.

File: en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+## Configure mutual SSL to establish secure connectivity
+
+To establish secure connectivity between the Choreo Gateway and your backend using mutual SSL, you must add the certificate of the backend (server certificate) to Choreo and add the certificate of Choreo (client certificate) as a trusted certificate in the backend.
+
+### Step 1: Obtain a certificate from Choreo
+
+There are two approaches you can take to get a certificate from Choreo.
+
+ - Generate a key pair with a self-signed certificate from Choreo, download the certificate, and install it in the backend.
```
[the remark this answers is not in the document]
Comment (Nashaath) [reply en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md]: @menakaj shall we avoid the term "install" and go with the following?
- Generate a key pair with a self-signed certificate from Choreo, download the public certificate, and add it to the backend.
- Use your own key pair to obtain a certificate from the Choreo Gateway and upload it. Subsequently, add the public certificate of this key pair to your backend.
WDYT?

**Review by menakaj — APPROVED** [review]
LGTM

**Review by menakaj — APPROVED** [review]

File: en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+Mutual SSL authentication involves both the client and server validating each other’s certificates before establishing a connection. The following diagram depicts this scenario:
+
+![Mutual SSL authentication](../assets/img/authentication-and-authorization/mutual-ssl-authentication.png)
+
+## Configure mutual SSL to establish secure connectivity
+
+To establish secure connectivity between the Choreo Gateway and your backend using mutual SSL, you must add the certificate of the backend (server certificate) to Choreo and add the certificate of Choreo (client certificate) as a trusted certificate in the backend.
+
+### Step 1: Configure TLS for a self-signed certificate
```
Comment (malinthaprasan) [inline en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md:13]: What we need to mention here is about setting up backend certificate right? Should we change the title accordingly?
`Configure backend service certificate`

File: en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+3. In the left navigation menu, click **Develop** and then click **Endpoints**.
+4. On the **Endpoints** page, click **Configure** corresponding to the endpoint.
+5. Click **Upload Endpoint Certificate**, and select the certificate file that you extracted in the prerequisites section to add it. This adds the certificate to all the environments as the default certificate for the endpoint. You can override this certificate if necessary when you deploy or promote the API.
+
+### Step 2: Obtain a certificate from Choreo
+
+There are two approaches you can take to get a certificate from Choreo.
+
+ - Generate a key pair with a self-signed certificate from Choreo, download the public certificate, and subsequently add and configure it in the backend.
+ - Use your own key pair to obtain a certificate from the Choreo Gateway and upload it. Subsequently, add and configure the public certificate of this key pair in your backend.
```
Comment (malinthaprasan) [inline en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md:41]: Upload your own public/private certificate pair to Choreo. Subsequently, add and configure the public certificate of this key pair in your backend.

File: en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+1. Sign in to the [Choreo Console](https://console.choreo.dev/).
+2. In the **Component Listing** pane, click on the API proxy for which you want to configure TLS. For instructions on how to create an API proxy component, see [Develop an API Proxy: Step 1](../develop-components/develop-an-api-proxy.md#step-1-create-an-api-proxy).
+3. In the left navigation menu, click **Develop** and then click **Endpoints**.
+4. On the **Endpoints** page, click **Configure** corresponding to the endpoint.
+5. Click **Upload Endpoint Certificate**, and select the certificate file that you extracted in the prerequisites section to add it. This adds the certificate to all the environments as the default certificate for the endpoint. You can override this certificate if necessary when you deploy or promote the API.
+
+### Step 2: Obtain a certificate from Choreo
+
+There are two approaches you can take to get a certificate from Choreo.
+
```
Comment (malinthaprasan) [inline en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md:39]: ### Step 2: Configure MTLS with the backend service

There are two approaches you can take to configure MTLS.

File: en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+### Step 1: Configure TLS for a self-signed certificate
+
+!!! info "Note"
+     This step is applicable only under the following conditions:
+
+       - Your backend uses a self-signed certificate.
+       - Your backend employs a certificate signed by an organization-wide certificate authority, which is not recognized as a well-known certificate authority, for enabling TLS.
+
+     If your backend uses valid CA-signed certificates, you can skip this step.
```
Comment (malinthaprasan) [inline en/docs/authentication-and-authorization/secure-communication-between-the-choreo-gateway-and-your-backend.md:21]: This is kind of an ideal situation but due to this issue https://github.com/wso2-enterprise/choreo/issues/28385 we cannot skip backend certificate configuration when configuring MTLS. I think we'll have to remove this note for now until we fix this issue.

**Comment by Nashaath** [comment]
Applied the suggested changes via https://github.com/wso2/docs-choreo-dev/pull/1459

**Pull request merged**